lib/rouge/lexers/brainfuck.rb

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Brainfuck < RegexLexer
+      tag 'brainfuck'
+      filenames '*.b', '*.bf'
+      mimetypes 'text/x-brainfuck'
+
+      title "Brainfuck"
+      desc "The Brainfuck programming language"
+
+      # optional comment or whitespace
+      ws = %r((?:\s|[^><\[\]\+\-\.,]*)+)
+      id = /[a-zA-Z_][a-zA-Z0-9_]*/
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          > < . , [ ] + -
+        )
+      end
+
+      def self.keywords_type
+        @keywords_type ||= Set.new %w(
+          
+        )
+      end
+
+      def self.reserved
+        @reserved ||= Set.new %w(
+          
+        )
+      end
+
+      def self.builtins
+        @builtins ||= Set.new %w(
+          
+        )
+      end
+
+      start { push :bol }
+
+      state :expr_bol do
+        mixin :inline_whitespace
+
+        rule /#if\s0/, Comment, :if_0
```

Comment:
This (and the one below) appears to be a "non-standard" extension, which may be fine, but the :if_0 state is empty? It's not clear what the intent is here, especially as `#if 0` is not included in the visual sample.